controllers/state_of_the_world.go

```diff
@@ -0,0 +1,139 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/kuadrant/policy-machinery/controller"
+	"github.com/kuadrant/policy-machinery/machinery"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	ctrlruntime "sigs.k8s.io/controller-runtime"
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+
+	kuadrantv1alpha1 "github.com/kuadrant/kuadrant-operator/api/v1alpha1"
+	kuadrantv1beta1 "github.com/kuadrant/kuadrant-operator/api/v1beta1"
+	kuadrantv1beta2 "github.com/kuadrant/kuadrant-operator/api/v1beta2"
+)
+
+var kuadrantNamespace = "kuadrant-system"
+
+//+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=gatewayclasses,verbs=list;watch
+
+func SetupWithManager(manager ctrlruntime.Manager, client *dynamic.DynamicClient) *controller.Controller {
```

Comment:
I got tricked by the name of this function. WDYT about renaming it to one of the following?
- `NewPolicyController`
- `NewPolicyMachineryController`
- `NewStateOfTheWorldReconciler`

Reply:
Ya I wasn't overly happy with the name my self. The `NewPolicyMachineryController` seem to be the best fit when looking at it here.